### PR TITLE
Update RISC-V readme.

### PR DESCRIPTION
**Pull request by rednoah91**

### CODE REVIEW

File: docs/get_started/getting_started_riscv_cmake.md [hunk 1]
```diff
@@ -35,21 +29,37 @@ $ ./build_tools/riscv/riscv_bootstrap.sh
...
+  -DIREE_BUILD_SAMPLES=OFF \
+  -DIREE_ENABLE_MLIR=OFF \
+  ..
 ```
 
 *   The above configures IREE to cross-compile towards `rv64` cpu platform.
 *   If user specify different download path (default in `${HOME}/riscv`) in `riscv_bootscrap.sh`, please append `-DRISCV_TOOL_PATH="/path/to/the/downloaded/folder"` in cmake command.
 
-### Build all targets
+build target:
```
Comment (hanhanW) [inline docs/get_started/getting_started_riscv_cmake.md:62]:
```suggestion
Build target:
```

**Review by hanhanW — APPROVED** [review]

File: docs/get_started/getting_started_riscv_cmake.md [hunk 2]
```diff
@@ -35,21 +29,37 @@ $ ./build_tools/riscv/riscv_bootstrap.sh
...
 
-### Configure on Linux
+### Host configuration
+
+Build and install at least the compiler tools on your host machine, or install them from a binary distribution:
+
+```shell
+$ mkdir build-host ; cd build-host ; mkdir install
+$ cmake -G Ninja -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=./install ..
+$ cmake --build . --target install
```
Comment (hcindyl) [inline docs/get_started/getting_started_riscv_cmake.md:39]: nit: I notice for Android, the host compiler is built outside of the iree root directory. Maybe for the sake of consistency, the shell script here can be the same? The host compiler should be shareable between Android and RiscV targets.

File: docs/get_started/getting_started_riscv_cmake.md [hunk 3]
```diff
@@ -35,21 +29,37 @@ $ ./build_tools/riscv/riscv_bootstrap.sh
...
+```
+
+Debugging note: if `IREE_LLVMAOT_LINKER_PATH` is set for targeting Android then the build above will fail, and you should run `unset IREE_LLVMAOT_LINKER_PATH`.
+
+### Target configuration
 
 ```shell
+# Back to iree root folder.
 $ mkdir build-riscv ; cd build-riscv
```
Comment (hcindyl) [inline docs/get_started/getting_started_riscv_cmake.md:48]: nit: Similar here. The build-riscv target can be outside of the iree root directory (use -B flag) so the build structure could be similar to other targets.

File: docs/get_started/getting_started_riscv_cmake.md [hunk 4]
```diff
@@ -35,21 +29,37 @@ $ ./build_tools/riscv/riscv_bootstrap.sh
...
 ```
 
 *   The above configures IREE to cross-compile towards `rv64` cpu platform.
 *   If user specify different download path (default in `${HOME}/riscv`) in `riscv_bootscrap.sh`, please append `-DRISCV_TOOL_PATH="/path/to/the/downloaded/folder"` in cmake command.
 
-### Build all targets
+build target:
 
 ```shell
 $ cmake --build .
```
Comment (hcindyl) [inline docs/get_started/getting_started_riscv_cmake.md:65]: nit: If -B is used during configuration, the build path needs to be updated too.

File: docs/get_started/getting_started_riscv_cmake.md [hunk 5]
```diff
@@ -35,21 +29,37 @@ $ ./build_tools/riscv/riscv_bootstrap.sh
...
+```
+
+Debugging note: if `IREE_LLVMAOT_LINKER_PATH` is set for targeting Android then the build above will fail, and you should run `unset IREE_LLVMAOT_LINKER_PATH`.
+
+### Target configuration
 
 ```shell
+# Back to iree root folder.
 $ mkdir build-riscv ; cd build-riscv
 $ cmake -G Ninja \
```
Comment (hcindyl) [inline docs/get_started/getting_started_riscv_cmake.md:49]: nit: Similar here. The build-riscv target can be outside of the iree root directory (use -B flag) so the build structure could be similar to other targets.

File: docs/get_started/getting_started_riscv_cmake.md (hunk 2 above)
Comment (hcindyl) [inline docs/get_started/getting_started_riscv_cmake.md:39]: nit: I notice for Android, the host compiler is built outside of the iree root directory. Maybe for the sake of consistency, the shell script here can be the same? The host compiler should be shareable between Android and RiscV targets.

**Review by hcindyl — COMMENTED** [review]
Sorry about the drop-in review. Please take a look of some minor suggestions.

File: docs/get_started/getting_started_riscv_cmake.md (hunk 2 above)
Comment (ScottTodd) [reply docs/get_started/getting_started_riscv_cmake.md:39]: +1, use `../iree-build-host/` as in the Android docs: https://google.github.io/iree/get-started/getting-started-android-cmake
```
$ cmake -G Ninja -B ../iree-build-host/ -DCMAKE_INSTALL_PREFIX=../iree-build-host/install .
$ cmake --build ../iree-build-host/ --target install
```

Directory structure is flexible, but we recommend putting build artifacts out of tree

recommended:
```
/dev/projects/iree/
/dev/projects/iree-build-host/ 
/dev/projects/iree-build-android/
```

should work, but not recommended by our docs:
```
/dev/projects/iree/
/dev/projects/iree/build-host/ 
/dev/projects/iree/build-android/
```

**Review by ScottTodd — CHANGES_REQUESTED** [review]

File: docs/get_started/getting_started_riscv_cmake.md (hunk 1 above)
Comment (rednoah91) [reply docs/get_started/getting_started_riscv_cmake.md:62]: OK

File: docs/get_started/getting_started_riscv_cmake.md (hunk 5 above)
Comment (rednoah91) [reply docs/get_started/getting_started_riscv_cmake.md:49]: OK

File: docs/get_started/getting_started_riscv_cmake.md (hunk 2 above)
Comment (rednoah91) [reply docs/get_started/getting_started_riscv_cmake.md:39]: Got it.

**Comment by rednoah91** [comment]
I'll rebase to avoid conflict with #4950

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**